File: Lab 12/EditorMemento.cs
public class EditorMemento
{
    public string Content{get;}
     public int CursorPosition { get; }
    public List<string> Selections { get; }
    public EditorMemento(string content, int cursorPosition, List<string> selections)
    {
        Content = content;
        CursorPosition = cursorPosition;
        Selections = new List<string>(selections);
    }
}

File: Lab_12_MementoPattarn/TextEditor/src/EditorMemento.java
import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

public class EditorMemento implements Serializable
{
    private static final long serialVersionUID=1;
    private final String content;
    private final int cursorPosition;

    private final List<String> selections;

    public EditorMemento(String content, int cursorPosition, List<String> selections) {
        this.content = content;
        this.cursorPosition = cursorPosition;
        this.selections = selections;
    }

    public String getContent() {
        return content;
    }
    public int getCursorPosition() {
        return cursorPosition;
    }

    public List<String> getSelections() {
        return new ArrayList<>(selections);
    }
}
